### PR TITLE
refactor: update `blas/ext/base/ssumkbn2` to follow current project conventions

### LINKED ISSUE #1542: [RFC]: refactor `blas/ext/base/ssumkbn2` to follow current project conventions
### Description

This RFC proposes refactoring [`@stdlib/blas/ext/base/ssumkbn2`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/ssumkbn2) to follow current project conventions.

These conventions are outlined in more detail in https://github.com/stdlib-js/stdlib/issues/788. Namely, we want to migrate from C++ add-on interfaces to C add-on interfaces and then make various style and simplification changes.

Some example commits which perform most, if not all, of the requested changes include https://github.com/stdlib-js/stdlib/commit/d66d80cfcbab1da95cb5ae8cfae4a1fc4c5129d1 (saxpy),  https://github.com/stdlib-js/stdlib/commit/3c82fe7b6c0f9fd6ef02c86d6e2024d708ac8c14 (cswap), and https://github.com/stdlib-js/stdlib/commit/5b15c9675be434c0e11b1ec1e1f78d7b089abbcd (dasum).

### Related Issues

Related issues #788, #1152.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by vr-varad**
## Description

> What is the purpose of this pull request?

This pull request refactors the ```@stdlib/blas/ext/ssumkbn2``` to follow current project conventions
> ## Checklist
> * [x]  update readme.md
> * [x]  docs
> * [x]  src
> * [x]  include.gypi
> * [x]  manifest.json
> * [x]  package.json
> * [x]  lib
> * [x]  examples
> * [x]  benchmark
> * [x]  test

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #1542 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> For previous context, please refer to discussion #1749.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by vr-varad** [comment]
@kgryte could u review this for any changes to be made.

**Comment by kgryte** [comment]
@vr-varad This one is looking good. I'll review within the next 24-48 hours.

File: lib/node_modules/@stdlib/blas/ext/base/ssumkbn2/benchmark/benchmark.js [hunk 1]
```diff
@@ -21,14 +21,19 @@
...
 var isnan = require( '@stdlib/math/base/assert/is-nan' );
 var pow = require( '@stdlib/math/base/special/pow' );
-var Float32Array = require( '@stdlib/array/float32' );
 var pkg = require( './../package.json' ).name;
 var ssumkbn2 = require( './../lib/ssumkbn2.js' );
 
 
+// VARIABLES //
+
+var rand = uniform( -20.0, -10.0 );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/ssumkbn2/benchmark/benchmark.js]:
```suggestion
var rand = uniform( -10.0, 10.0 );
```

File: lib/node_modules/@stdlib/blas/ext/base/ssumkbn2/benchmark/benchmark.native.js [hunk 2]
```diff
@@ -36,6 +36,7 @@ var ssumkbn2 = tryRequire( resolve( __dirname, './../lib/ssumkbn2.native.js' ) )
 var opts = {
 	'skip': ( ssumkbn2 instanceof Error )
 };
+var rand = uniform( -20.0, -10.0 );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/ssumkbn2/benchmark/benchmark.native.js]:
```suggestion
var rand = uniform( -10.0, 10.0 );
```

File: lib/node_modules/@stdlib/blas/ext/base/ssumkbn2/benchmark/benchmark.ndarray.js [hunk 3]
```diff
@@ -21,14 +21,19 @@
...
 var isnan = require( '@stdlib/math/base/assert/is-nan' );
 var pow = require( '@stdlib/math/base/special/pow' );
-var Float32Array = require( '@stdlib/array/float32' );
 var pkg = require( './../package.json' ).name;
 var ssumkbn2 = require( './../lib/ndarray.js' );
 
 
+// VARIABLES //
+
+var rand = uniform( -20.0, -10.0 );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/ssumkbn2/benchmark/benchmark.ndarray.js]:
```suggestion
var rand = uniform( -10.0, 10.0 );
```

File: lib/node_modules/@stdlib/blas/ext/base/ssumkbn2/benchmark/benchmark.ndarray.native.js [hunk 4]
```diff
@@ -36,6 +36,7 @@ var ssumkbn2 = tryRequire( resolve( __dirname, './../lib/ndarray.native.js' ) );
 var opts = {
 	'skip': ( ssumkbn2 instanceof Error )
 };
+var rand = uniform( -20.0, -10.0 );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/ssumkbn2/benchmark/benchmark.ndarray.native.js]:
```suggestion
var rand = uniform( -10.0, 10.0 );
```

File: lib/node_modules/@stdlib/blas/ext/base/ssumkbn2/lib/ndarray.native.js [hunk 5]
```diff
@@ -37,20 +38,19 @@ var addon = require( './ssumkbn2.native.js' );
...
 	var view;
-	if ( stride < 0 ) {
-		offset += (N-1) * stride;
-	}
-	view = new Float32Array( x.buffer, x.byteOffset+(x.BYTES_PER_ELEMENT*offset), x.length-offset ); // eslint-disable-line max-len
+
+	offset = minViewBufferIndex( N, stride, offset );
+
+	view = offsetView( x, offset );
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/ssumkbn2/lib/ndarray.native.js]:
```suggestion
	offset = minViewBufferIndex( N, stride, offset );
	view = offsetView( x, offset );
```

File: lib/node_modules/@stdlib/blas/ext/base/ssumkbn2/manifest.json [hunk 6]
```diff
@@ -1,40 +1,45 @@
...
-				"./include"
-			],
-			"libraries": [
-				"-lm"
-			],
-			"libpath": [],
-			"dependencies": []
-		}
-	]
+  "options": {},
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/ssumkbn2/manifest.json]: This file has mixed TAB and space indentation. It should be 2-space indentation.

Additionally, it needs additional configuration objects. See, e.g., https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/blas/ext/base/dfill/manifest.json

File: lib/node_modules/@stdlib/blas/ext/base/ssumkbn2/src/addon.c [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+* @return       Node-API value
+*/
+static napi_value addon( napi_env env, napi_callback_info info ) {
+	STDLIB_NAPI_ARGV( env, info, argv, argc, 3 );
+	STDLIB_NAPI_ARGV_INT64( env, N, argv, 0 );
+	STDLIB_NAPI_ARGV_INT64( env, stride, argv, 2 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, X, N, stride, argv, 1 );
+
+	napi_value v;
+	napi_status status = napi_create_double( env, (double)stdlib_strided_ssumkbn2( N, (float *)X, stride ), &v );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/ssumkbn2/src/addon.c]:
```suggestion
	napi_status status = napi_create_double( env, (double)stdlib_strided_ssumkbn2( N, X, stride ), &v );
```

File: lib/node_modules/@stdlib/blas/ext/base/ssumkbn2/src/addon.c [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+	assert( status == napi_ok );
+
+	return v;
+}
+
+STDLIB_NAPI_MODULE_EXPORT_FCN( addon )
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/ssumkbn2/src/addon.c]:
```suggestion
STDLIB_NAPI_MODULE_EXPORT_FCN( addon )

```

**Review by kgryte — CHANGES_REQUESTED** [review]
Thanks for working on this @vr-varad. Looks like the main blocker is just updating the `manifest.json` file. Once that is resolved, should be good to merge.

**Comment by vr-varad** [comment]
Hello @kgryte, I've addressed the issues. Would you mind reviewing it when you have a moment? Thank you for your time!

**Comment by vr-varad** [comment]
Hello @kgryte, I've addressed the issues. Would you mind reviewing it when you have a moment? Thank you for your time!

File: lib/node_modules/@stdlib/blas/ext/base/ssumkbn2/lib/ndarray.native.js [hunk 9]
```diff
@@ -37,20 +38,17 @@ var addon = require( './ssumkbn2.native.js' );
...
+* var v = ssumkbn( 4, x, 2, 1 );
 * // returns 5.0
 */
 function ssumkbn2( N, x, stride, offset ) {
 	var view;
-	if ( stride < 0 ) {
-		offset += (N-1) * stride;
-	}
-	view = new Float32Array( x.buffer, x.byteOffset+(x.BYTES_PER_ELEMENT*offset), x.length-offset ); // eslint-disable-line max-len
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/ssumkbn2/lib/ndarray.native.js]:
```suggestion
```

File: lib/node_modules/@stdlib/blas/ext/base/ssumkbn2/manifest.json [hunk 10]
```diff
@@ -1,40 +1,76 @@
 {
-	"options": {},
+	"options": {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/ssumkbn2/manifest.json]: This file has mixed TAB and space indentation. It should be strictly 2-space indentation.

File: lib/node_modules/@stdlib/blas/ext/base/ssumkbn2/manifest.json [hunk 11]
```diff
@@ -1,40 +1,76 @@
...
+		],
+		"libraries": [
+		  "-lm"
+		],
+		"libpath": [],
+		"dependencies": []
+	  }
 	]
-}
+  }
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/ssumkbn2/manifest.json]:
```suggestion
  }

```

File: lib/node_modules/@stdlib/blas/ext/base/ssumkbn2/manifest.json [hunk 12]
```diff
@@ -1,40 +1,76 @@
 {
-	"options": {},
+  "options": {
+    "task": "build"
+	},
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/ssumkbn2/manifest.json]: This file still has mixed space and TAB indentation. :)

**Comment by kgryte** [comment]
@vr-varad What editor are you using? Can you not inspect the indentation locally? Each commit you've made attempting to address the indentation in that file has not actually fixed the indentation.

**Comment by vr-varad** [comment]
@kgryte I am using VSCODE as the editor and I am using tab size at 2 space indendation

**Comment by vr-varad** [comment]
for the last commit I did it manually but don't know why its not working

**Comment by kgryte** [comment]
You want to indent using spaces, not tabs, for that file. You should be able to run a command in VSCode to convert all tabs to spaces.

**Comment by vr-varad** [comment]
ya I got it Thanks @kgryte

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @vr-varad!

**Pull request merged**